wikimedia/wikipedia-ios
# Display Edit notices if available

**Phabricator:** https://phabricator.wikimedia.org/T201596

### Notes
This adds the presentation of the Edit notices modal (if Edit notices are available) when the user edits an article. 

Some outstanding items:
- While in my testing so far on device it's uncommon that the section editor's keyboard toolbar overlaps the modal that's presented, I think this is still a corner case we may need to address
- Still need to connect to the section editor dismiss confirmation action sheet when tapping a URL

### Test Steps
It's best to start from a delete/fresh install to confirm the UserDefaults for the editing welcome controller are reset.

1. Find an article with Edit notices (I often use `Covid-19`)
2. Tap to edit a section and confirm you see the editing welcome controller
3. After completing the editing welcome, confirm you are presented with the section editor and a modal appears with any edit notices (note, you can also alter `EditNoticesFetcher` to remove the filtering by `editnotice` to see all available notices displayed here).
4. Toggle the switch and return to step 1. Confirm editing again does not automatically display the article, but adds a bar button to display the modal again.
5. Try changing the app's theme and confirm the modal still renders as expected.
6. Tap a link in the modal and confirm it dismisses the modal and editor then pushes the URL over the article view controller.
